CDCgov/prime-reportstream
# 9009 Moved NTE (AOE notes) between the OBX and SPM segments

This PR adds additional CA receiver transforms to move all notes to the last observation segment in an HL7 v2 message. CA only looks at notes that are between the the last OBX and SPM segments.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Send the attached FHIR bundle to the Reports API:
[Simple Report.notes test.txt](https://github.com/CDCgov/prime-reportstream/files/11213524/Simple.Report.notes.test.txt)


2. Verify that the request was accepted

3. Open the generated HL7 file in the local sftp folder.

4. Verify NTE segments are mapped between the last OBX and SPM segments.
 
5. Run integration tests and make sure they pass

## Changes
- Added CA receiver transforms
- Updated integration tests

## Checklist

### Testing

### Process

## Linked Issues
- #9009 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
